docs-src/concepts.md

```diff
@@ -230,7 +230,24 @@ be differences in behavior from the spec.
 
 ### Extension points
 
-TODO
+A number of extension points are provided in the API to provide flexibility in
+addressing the large number of use-cases that cannot be addressed by a general
+purpose API.
+
+Here is a summary of extension points in the API:
+- **Custom Routes**: Implementers can chose to create custom Route resources
+  for protocols that are not currently supported in the API.
```

Comment:
Since I think this intended to be our extension point of last resort, maybe we should list it last with some kind of caveat like "if none of the above extension points are sufficient, you can create a custom route resource..."

Reply:
Totally makes sense, updated.